idaes/core/util/model_diagnostics.py

```diff
@@ -4122,3 +4264,269 @@ def _collect_model_statistics(model):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+        val = -self._get_value_for_sum_subexpression(child_data[0])
+
+        return [val], mismatch, cancelling, const
+
+    def _check_abs(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+        val = abs(self._get_value_for_sum_subexpression(child_data[0]))
+
+        return [val], mismatch, cancelling, const
+
+    def _check_unary_function(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+
+        func_name = node.getname()
+        func = getattr(math, func_name)
+        func_val = self._get_value_for_sum_subexpression(child_data[0])
+
+        try:
+            val = func(func_val)
+        except ValueError:
+            raise ValueError(
+                f"Error in ConstraintTermAnalysisVisitor: error evaluating {str(node)}."
+            )
+
+        return [val], mismatch, cancelling, const
+
+    def _check_other_expression(self, node, child_data):
+        mismatch, cancelling, const = self._perform_checks(node, child_data)
+
+        # First, need to get value of input terms, which may be sub-expressions
+        input_mag = [self._get_value_for_sum_subexpression(i) for i in child_data]
+
+        # Next, create a copy of the external function with expected magnitudes as inputs
+        newfunc = node.create_node_with_local_data(input_mag)
+
+        # Evaluate new function and return the absolute value
+        return [value(newfunc)], mismatch, cancelling, const
+
+    def _perform_checks(self, node, child_data):
+        # Perform checks for problematic expressions
+        # First, need to check to see if any child data is a list
+        # This indicates a sum expression
+        mismatch = []
+        cancelling = []
+        const = True
+
+        for d in child_data:
+            # Collect all warnings about mismatched terms from child nodes
+            mismatch += d[1]
+            cancelling += d[2]
+
+            # We will check for cancelling terms here, rather than the sum itself, to handle special cases
+            # We want to look for cases where a sum term results in a value much smaller
+            # than the terms of the sum
+            sums = self._sum_combinations(d[0])
+            if any(i <= self._sum_tol * max(d[0]) for i in sums):
+                cancelling.append(str(node))
+
+            # Expression is not constant if any child is not constant
+            if not d[3]:
+                const = False
+
+        # Return any problematic terms found
+        return mismatch, cancelling, const
+
+    def _check_equality_expression(self, node, child_data):
+        # (In)equality expressions are a special case of sum expressions
+        # We can start by just calling the method to check the sum expression
+        vals, mismatch, cancelling, const = self._check_sum_expression(node, child_data)
```

Comment:
Don't you need to negate the values for the second argument in the relational expression before treating it like a sum?

Reply:
I had forgotten to make that correction - thank you.